src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Ported from https://github.com/adafruit/Adafruit_Python_BMP/blob/master/Adafruit_BMP/BMP085.py  
+// Formulas and code examples can also be found in the datasheet https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+
+using System;
+using System.Device.I2c;
+using System.Buffers.Binary;
+using System.Threading;
+using Iot.Units;
+
+namespace Iot.Device.Bmp180
+{
+    public class Bmp180 : IDisposable
+    {
+        private I2cDevice _i2cDevice;        
+        private readonly CalibrationData _calibrationData;
+        private Sampling _mode;
+        public const byte DefaultI2cAddress = 0x77;
+
+        public Bmp180(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;            
+            _calibrationData = new CalibrationData();            
+            //Read the coefficients table
+            _calibrationData.ReadFromDevice(this);
+            SetSampling(Sampling.Standard);
+        }
+        
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sampling Mode</param>
+        public void SetSampling(Sampling mode)
+        {
+            _mode = mode;
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature ReadTemperature()
+        {
+            int B5 = CalculateTrueTemperature();
+            int T = (B5 + 8) / (1 << 4);

```

Comment:
wouldn't it be better to shift right instead? or even better just divide by `160.0` so you get more accurate result?

Reply:
> 
> 
> .... or even better just divide by `160.0` ...

Do you mean `16.0 = (1 << 4)` or `160.0`?

I know that the **shift operation** looks like a code for a microcontroller 🤖  ,  but its like appear in the datasheet. 

I can change all the **shift operation** for equivalent `int` value.

Reply:
`1 << 4 = 16` but then you later divide by another 10  and then cast to double (by constructing Temperature) so effectively you divide by 160. While doing that you don't need to lose precision if you divide by 160.0 (which is double - dividing by integer removes fractions)

Reply:
I got it. I will try simplify all the logic.